Framework/PythonInterface/plugins/algorithms/IntegratePeaks1DProfile.py

```diff
@@ -0,0 +1,794 @@
+                    fwhm = get_fwhm_from_back_to_back_params(peak, ws, detid)
+                    nbins = max(3, int(nfwhm * fwhm / bin_width)) if fwhm is not None else self.getProperty("NBins").value
+                else:
+                    nbins = self.getProperty("NBins").value
+
+                # get data array and crop
+                peak_data = array_converter.get_peak_data(peak, detid, bank_name, nrows, ncols, nrows_edge, ncols_edge)
+
+                # fit peak
+                peak_fitter = PeakFitter(
+                    ws,
+                    peaks,
+                    ipk,
+                    peak,
+                    peak_data,
+                    nbins,
+                    error_strategy,
+                    weight_func,
+                    cost_func,
+                    peak_func_name,
+                    bg_func,
+                    peak_params_to_fix,
+                    frac_dspac_delta,
+                    i_over_sig_threshold,
+                )
+                successful, attempted, yfits, intens_sum, sigma_sq_sum = peak_fitter.integrate_peak()
+
+                # update intenisty and save results for later plotting
+                if np.any(successful):
+                    # check edge
+                    is_on_edge = np.any(np.logical_and(attempted, peak_data.det_edges))
+                    if integrate_on_edge or not is_on_edge:
+                        status = PEAK_STATUS.VALID
+                        intens = intens_sum
+                        sigma = np.sqrt(sigma_sq_sum)
+                    else:
+                        status = PEAK_STATUS.ON_EDGE
+                    if output_file:
+                        intens_over_sig = intens / sigma if sigma > 0 else 0.0
+                        results.append(
+                            LineProfileResult(
+                                ipk,
+                                peak,
+                                intens_over_sig,
+                                peak_fitter.tofs,
+                                peak_fitter.y,
+                                peak_fitter.esq,
+                                yfits,
+                                successful,
+                                peak_data.irow,
+                                peak_data.icol,
+                                status,
+                            )
+                        )
+
+            set_peak_intensity(peak, intens, sigma, do_lorz_cor)
+
+        # plot output
+        if output_file:
+            prog_reporter.resetNumSteps(len(results), start=0.0, end=1.0)
+            plot_integration_results(output_file, results, prog_reporter)
+
+        # assign output
+        self.setProperty("OutputWorkspace", peaks)
+
+    def exec_child_alg(self, alg_name, **kwargs):
+        alg = self.createChildAlgorithm(alg_name, enableLogging=False)
+        alg.initialize()
+        for prop, value in kwargs.items():
+            alg.setProperty(prop, value)
```

Comment:
I think this can be simplified
```suggestion
        alg.setProperties(kwargs)
```

Reply:
Thanks - I didn't know about this!